Server/src/main/java/com/gluonhq/picluster/server/TaskQueue.java

```diff
@@ -21,6 +21,13 @@ static void add(Task task) {
         logger.fine("Notified all waiters");
     }
 
+    static boolean containsTaskWithId(String id) {
+        synchronized (queue) {
+            Optional<Task> opt = queue.stream().filter(t -> !t.id.equals(id)).findFirst();
```

Comment:
`filter()` is a "positive" method meaning that any items that match the given predicate will be kept in the stream as opposed to be filtered out. For this pattern, `anyMatch()` is a simpler alternative:

    return queue.stream().anyMatch(t -> t.id.equals(id));